channels/channel.no/nrkno/chn_nrkno.py

```diff
@@ -320,48 +321,58 @@ def create_category_sub_item(self, result_set: dict) -> Optional[Union[List[Medi
         #
         # return self.create_generic_item(result_set, program_type)
 
-    def create_episode_item(self, result_set):
-        """ Creates a MediaItem of type 'folder' using the result_set from the regex.
+    def create_search_item(self, result_set):
+        """ Creates a MediaItem of type 'folder' for a search result using the result_set
+        from the regex.
 
         This method creates a new MediaItem from the Regular Expression or Json
         results <result_set>. The method should be implemented by derived classes
         and are specific to the channel.
 
-        :param list[str]|dict[str,str] result_set: The result_set of the self.episodeItemRegex
+        :param list[str]|dict[str,Any] result_set: The result_set of the self.episodeItemRegex
 
-        :return: A new MediaItem of type 'folder'.
+        :return: A new MediaItem of type 'folder' or 'video'.
         :rtype: MediaItem|None
 
         """
 
         title = result_set["title"]
+        content_id = result_set["contentId"]
 
-        program_type = result_set.get("type", "???").lower()
-        if program_type not in ("programme", "series"):
-            Logger.debug("Item '%s' has type '%s'. Ignoring", title, program_type)
-            return None
+        if result_set["type"] == "series":
+            url = "https://psapi.nrk.no/tv/catalog/series/{}?apiKey={}".format(content_id, self.__api_key)
+            item = FolderItem(title, url, content_type=contenttype.TVSHOWS)
+        else:
+            url = self.__get_video_url(content_id)
+            item = MediaItem(title, url, media_type=mediatype.VIDEO)
 
-        return self.create_generic_item(result_set, program_type)
+        self.__set_image(item, result_set["webImages"], "width", "uri", False)
 
-    def create_search_item(self, result_set):
-        """ Creates a MediaItem of type 'folder' for a search result using the result_set
-        from the regex.
+
+        return item
+
+    def create_episode_item(self, result_set):
```

Comment:
Any reason why you swapped `create_episode_item` and the `create_search_item`?

Reply:
This order felt more natural as both `create_episode_item` and `create_video_item` use `create_generic_item`, which `create_search_item` doesn't do anymore.